bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/RawMirrorRequest.java

```diff
@@ -88,19 +92,25 @@ public IArtifactDescriptor getArtifactDescriptor() {
 	// Perform the mirror operation without any processing steps
 	@Override
 	protected IStatus getArtifact(IArtifactDescriptor artifactDescriptor, OutputStream destination, IProgressMonitor monitor) {
-		if (SimpleArtifactRepository.CHECKSUMS_ENABLED) {
-			Collection<ChecksumVerifier> steps = ChecksumUtilities.getChecksumVerifiers(artifactDescriptor,
-					IArtifactDescriptor.DOWNLOAD_CHECKSUM, Collections.emptySet());
-			if (steps.isEmpty()) {
-				LogHelper.log(new Status(IStatus.WARNING, Activator.ID,
-						NLS.bind(Messages.noDigestAlgorithmToVerifyDownload, artifactDescriptor.getArtifactKey())));
+		monitor = IProgressMonitor.nullSafe(monitor);
```

Comment:
`SubMonitor.convert` can handle null monitors and can't we just call done on the sub-monitor?
Again I think the try-catch is not necessary.

Reply:
all my concerns are expressed by the junit result.

Reply:
But asserting in AbstractProvisioningUITest that for example `ProvisioningSession.performProvisioningPlan()` consumes all work of the monitor is not right.

From the Java-doc of IProgressMonitor:
https://github.com/eclipse-equinox/equinox/blob/8beb9ce0b1dea1176b06ad7300f2663ff6db4eff/bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/IProgressMonitor.java#L57

```
CALLER/CALLEE RESPONSIBILITIES: 

Methods that receive an {@link IProgressMonitor} ("callees") must either obey the following conventions or include  these rules. The called method:
[...]
- Will not promise to invoke {@link #done()} on the monitor.

[...]
The caller:
[...]
Will not rely on the callee to invoke {@link #done()} on the monitor. It must either select an implementation of {@link IProgressMonitor} that does not require {@link #done()} to be called, for example {@link SubMonitor}, or  it must invoke {@link #done()} itself after the method returns.
```

And the only real caller of `ProvisioningSession.performProvisioningPlan()` in ProfileModificationJob.runModal() even explicitly calls done, although even that would actually not be necessary since the caller of that method respectively the creator of a progress-monitor is responsible to call done.

So asserting that in the tests is too strict.